icicle/src/device_api.cpp

```diff
@@ -58,6 +58,16 @@ namespace icicle {
 
     const Device& get_default_device() { return m_default_device; }
 
+    eIcicleError set_default_device(const Device& dev)
+    {
+      if (!is_device_registered(dev.type)) {
+        THROW_ICICLE_ERR(
+          eIcicleError::INVALID_DEVICE, "Device type " + std::string(dev.type) + " has not been registered");
+      }
+      m_default_device = dev;
```

Comment:
note that this is not thread safe. Not sure if it's a problem though, since users are not supposed to call this from multiple threads.

Reply:
I will add docs for this and will mention it there